### PR TITLE
get proc usage concurrently

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: avm/src/daemon/daemon.rs [hunk 1]
```diff
@@ -100,20 +101,27 @@ async fn post_v1_scale(cluster_id: &str, agz_b64: &str, deploy_token: &str, fact
...
+        },
+        Err(_) => 0.0
+      }
+    });
+  let mut futures = Vec::new();
+  while let Some(fut) = stream.next().await {
+    futures.push(fut);
   }
-  usages
+  join_all(futures).await
```
Comment (cdmistman) [inline avm/src/daemon/daemon.rs:124]: I believe this can all be done with `.collect().await`

Comment (depombo) [reply avm/src/daemon/daemon.rs:124]: What would the type of `collect` be?

Comment (cdmistman) [reply avm/src/daemon/daemon.rs:124]: `Vec<f32>`

Comment (cdmistman) [reply avm/src/daemon/daemon.rs:124]: turns out it'd be `Vec<impl Future<Output = f32>>`, so you do still need the `join_all`

Comment (depombo) [reply avm/src/daemon/daemon.rs]: yes, thanks for the help!

**Review by cdmistman — APPROVED** [review]

File: avm/src/daemon/daemon.rs [hunk 2]
```diff
@@ -100,20 +101,25 @@ async fn post_v1_scale(cluster_id: &str, agz_b64: &str, deploy_token: &str, fact
...
-      // does not represent any reasonable value by itself
-      sleep(Duration::from_secs(1)).await;
-      let measurement_2 = proc.cpu_usage().await.expect("Failed to get CPU usage for process");
-      let usage = (measurement_2 - measurement_1).get::<ratio>();
-      usages.push(usage);
-    }
-  }
-  usages
+  let processes = processes();
+  let futures = processes
```
Comment (dfellis) [inline avm/src/daemon/daemon.rs:105]: Nit: can it just be `let futures = processes()...`?

**Review by dfellis — APPROVED** [review]
Looks great!

**Pull request merged**